modules/supporting_encapsulation.adoc

```diff
@@ -0,0 +1,93 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/optimizing-cpu-usage.adoc
+
+[id="supporting-encapsulation_{context}"]
+= Supporting encapsulation
+
+As an OpenShift support engineer or developer, you want to inspect Kubernetes-specific mountpoints.
+
+== Locating the encapsulated namespaces
+
+Any monitoring tool that relies on the ability to run in the host OS and have visibility of mountpoints created by Kubelet, the Container Runtime, or containers themselves, will need to enter the container mount namespace in order to see these mountpoints. The provided `kubensenter` script will execute another command inside the Kubernetes mountpoint and can be used to adapt any existing tools.
+The `kubensenter` script is aware of the state of the mount encapsulation feature status, and is safe to run even if encapsulation is not enabled. In that case it will simply execute the provided command in the default mount namespace.
+For example, if a systemd service needs to run inside the new Kubernetes mount namespace, you should edit the service file and prefix the `ExecStart=` command line with `kubensenter`.
+
+[source,terminal]
+----
+[Unit]
+Description=Example service
+[Service]
+ExecStart=/usr/bin/kubensenter /path/to/original/command arg1 arg2
+----
+
+
+== Inspecting the encapsulated namespaces
+
+An administrator logged in to the host OS wishing to obtain information about the Kubernetes system for debugging or auditing purposes will need to be aware of the new mount namespace. A shell originating in a container, such as `oc debug` will already be inside the Kubernetes namespace. A shell originating in an SSH session will be in the default namespace, and an administrator will need to run the `kubensenter` script as root to view or interact with these mountpoints.
```

Comment:
So you only need this kubensenter script when SSHing into the cluster? not `oc debug`?

Reply:
Correct.  Because the `oc debug` shell is in a container, it is already inside the Kubernetes mountpoint.

However, for ease in scripting it's fine to always run it - If you're already inside the namespace there's no harm in running it again, and if the feature is disabled there's also no harm in running it (in that it does nothing).

Reply:
Will discuss with Jim.